libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1029,7 +1029,7 @@ string ABIFunctions::abiEncodingFunctionFunctionType(
 )
 {
 	solAssert(_from.kind() == FunctionType::Kind::External, "");
-	solAssert(_from == _to, "");
+	solAssert(_from.isImplicitlyConvertibleTo(_to), "");
```

Comment:
This is very likely a no-op, but to be safe, I think it is better to also call the conversion function in that case.

Reply:
(i.e. just call it in all cases)